src/rpc/blockchain.cpp

```diff
@@ -1640,6 +1640,37 @@ static T CalculateTruncatedMedian(std::vector<T>& scores)
     }
 }
 
+static constexpr int NUM_GETBLOCKSTATS_PERCENTILES = 5;
+
+template<typename T>
+static void CalculatePercentilesByWeight(T result[NUM_GETBLOCKSTATS_PERCENTILES], std::vector<std::pair<T, int64_t>>& scores, int64_t total_weight)
+{
+    const size_t size = scores.size();
+    if (size == 0) {
```

Comment:
You don't use `size` anywhere else below - so you could just do:

    if (scores.empty()) {
        return;
    }

Reply:
:+1: